tests/test_authenticator.py

```
@@ -2975,7 +2975,15 @@ def test_basic_auth_temp_token_returns_existing(self):
                 another_token = another_response.json.get('access_token')
                 assert another_token == token
 
-    def test_basic_auth_temp_token_returns_new_token(self):
+
+    @pytest.mark.parametrize(
+        'delta',
+        [
+            pytest.param(61, id='after_one_minute'),
+            pytest.param(3601, id='after_one_hour')
```

Shouldn't the authenticator return an error if trying to retrieve after more than one hour? Or does changing the `frozen_time` reset things in a way that I'm not following?

---

You're right that an error is raised, I think it does it [here](https://github.com/NYPL-Simplified/circulation/blob/develop/api/authenticator.py#L825). I don't think this is a good/accurate test now, I'll revert it.

---

okay, I see, so the check for an expired token happens there, _before_ we get to this method. So we can assume that this method is only dealing with the renewal of a valid token or creation of a new one. Makes sense to eliminate this test here then, unless we want to make sure that this also guards against that.

---

I think this test was fine. A controller gated by an authentication token should raise an error when you send it an expired token (e.g. one more than an hour old), but this controller isn't of that type. This controller doesn't take a token for authentication, it only takes username/password. It uses that information to _issue_ a token, and the test is verifying that if the token already in the database is a _minute_ old, this controller issues a new token instead of recycling the old one.